client/src/pages/course/student/auto-test.tsx

```diff
@@ -51,6 +51,22 @@ function Page(props: CoursePageProps) {
   const [verifications, setVerifications] = useState<Verification[]>([]);
   const [courseTaskId, setCourseTaskId] = useState(null as number | null);
 
+  const [isModified, setIsModified] = useState(false);
+
+  const handleClosePage = (e: BeforeUnloadEvent) => {
+    e.preventDefault();
+    e.returnValue = 'Your answers will be lost! Do you really want to exit this page?';
+  };
+
+  useEffect(() => {
+    if (isModified) {
+      window.addEventListener('beforeunload', handleClosePage);
```

Comment:
Please consider to use `useBeforeUnload` from 'react-use` library. We have it in dependencies